### PR TITLE
[NOREF] - Filter view view route restructure

### PR DESCRIPTION
**Pull request by patrickseguraoddball**
# NOREF

## Changes and Description

- Fixed app crash when passing an unmapped filter query param
- Reworked readonly routing to all for filter views to be passed without a page subkey

This is in preparation for the BE to begin supplying filter view links to readonly without needing to specify a subpage

Ex: `${host}/models/${modelID}/read-only`
Ex: `${host}/models/${modelID}/read-only?filter-view=ccw`

Previous the routing didn't allow for filter views without a specified subpage (`model-basics`, etc)

## How to test this change

Test out navigation to `${host}/models/${modelID}/read-only` without a subpage , should redirect to `model-basics`
Append a filter-view query param to `${host}/models/${modelID}/read-only` and verify it renders - `${host}/models/${modelID}/read-only?filter-view=ccw`
Try to type in an incorrect filter-view param - should reroute back to model basics

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

File: src/views/ModelPlan/ReadOnly/_components/FilterView/Banner/index.tsx [hunk 1]
```diff
@@ -54,7 +56,7 @@ const FilterViewBanner = ({
                 type="button"
                 unstyled
                 className="text-white text-no-wrap"
-                onClick={() => history.push(`${history.location.pathname}`)}
+                onClick={() => history.push(history.location.pathname)}
```
Comment (ClayBenson94) [inline src/views/ModelPlan/ReadOnly/_components/FilterView/Banner/index.tsx:59]: Was this a functional change, or just stylistic? Only case I could see this mattering is pushing `undefined` or `null` vs `"undefined"` or `"null"`

File: src/views/App/index.tsx [hunk 2]
```diff
@@ -104,14 +104,8 @@ const AppRoutes = () => {
...
 
-      <Redirect
-        exact
-        from="/models/:modelID/read-only"
-        to="/models/:modelID/read-only/model-basics"
-      />
-
       <SecureRoute
-        path="/models/:modelID/read-only/:subinfo"
+        path="/models/:modelID/read-only/:subinfo?"
```
Comment (ClayBenson94) [inline src/views/App/index.tsx:108]: What does the addition of `?` do to this route?

Comment (patrickseguraoddball) [reply src/views/App/index.tsx:108]: The `?` makes the parameter optional.  The component can still be rendered without passing any of the subinfo (`model-basics`, `payments`, etc)

File: src/views/ModelPlan/ReadOnly/index.tsx [hunk 3]
```diff
@@ -260,10 +267,14 @@ const ReadOnly = ({ isHelpArticle }: { isHelpArticle?: boolean }) => {
   const subComponent = subComponents[subinfo];
 
   if ((!loading && error) || (!loading && !data?.modelPlan)) {
-    return <NotFoundPartial />;
+    return <NotFound />;
+  }
+
+  if (!subinfo && !isViewingFilteredGroup) {
+    history.replace(`${location.pathname}/model-basics`);
```
Comment (ClayBenson94) [inline src/views/ModelPlan/ReadOnly/index.tsx:274]: Nit: Might be nice to pull `model-basics` into a `const` at the top of this component under something like

```typescript
const defaultSection = "model-basics"
```

just for visibility.

**Review by ClayBenson94 — CHANGES_REQUESTED** [review]
Requesting changes for the default section comments

File: src/views/ModelPlan/ReadOnly/_components/FilterView/Banner/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (patrickseguraoddball) [reply src/views/ModelPlan/ReadOnly/_components/FilterView/Banner/index.tsx:59]: Reply

File: src/views/App/index.tsx (hunk 2 above)
Comment (patrickseguraoddball) [reply src/views/App/index.tsx:108]: test

**Review by ClayBenson94 — CHANGES_REQUESTED** [review]
Not sure how important this really is, but going to `http://localhost:3005/models/92b151fc-3cdc-4999-ae4d-abfe8503603b/read-only/` (note the trailing slash) ends up booting me to `http://localhost:3005/models/92b151fc-3cdc-4999-ae4d-abfe8503603b/read-only//model-basics`, which gives me a "Page not found"

Not sure there's an easy way to fix this, but could be a headache for folks that chop off the section in the URL themselves 🤷

**Review by ClayBenson94 — APPROVED** [review]

**Pull request merged**